recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -165,6 +172,24 @@ add_task(withMockExperiments(withMockPreferences(async function (experiments, mo
   startObserver.restore();
 })));
 
+// start should detect if a new preference value type matches the previous value type
+add_task(withMockPreferences(async function (mockPreferences) {
+  mockPreferences.set("fake.type_preference", "oldvalue");
+
+  await Assert.rejects(
+    PreferenceExperiments.start({
```

Comment:
We should also have a test that the forwarded (wrapAsync) stuff on the NormandyDriver throws, too (so we don't accidentally swallow exceptions there).

More generally, we should make sure that the start() callers do the right thing with these exceptions. I tried to find them but github search is awful and this all seems to still live on a branch, maybe / sort of? So I couldn't figure out where the entrypoint is... 

In any case, we should make sure that if starting one of these experiments fails, it doesn't break running any/all experiments. Especially because, for preferences that don't live in modules/libpref/init/all.js and friends (ie appear in about:config on a default profile), it's possible for users to define those prefs with an unexpected type, and we wouldn't want users doing that to be breaking their browsers and/or all our code.